dogfood.md

```diff
@@ -0,0 +1,167 @@
+
+## Clients
+
+The memeolist dogfood exercise will support iOS and Android native as well as cross platform development with our Cordova and Xamarin SDKs.  Additionally, the use of GraphQL should let us generate some of the client code which means we won't be starting from scratch for each platform.
+
+### Android
+
+David Martin has done some work with graphQL and Android related to the current research on a new sync service.  The Apollo project has good tooling for supporting GraphQL APIs on Android, and the AeroGear Android SDK supports all of the features we are planning on targetting.  The existing memeolist code from the FeedHenry demo can also serve as a starting point for development and testing, though it will need to undergo a lot of revisions.
+
+### Xamarin SDK
+
+The Xamarin SDK is lacking support for UPS which will have to be implemented for both iOS and Android.  Additionally we lack experience with Xamarin and GraphQL, but a Google search shows that there are a few projects which can be researched for support.  This application will also offer people interested in upskilling on Xamarin a chance to work on a project from greenfield.
+
+### Cordova SDK
+GraphQL seems to have a lot of javascript support which will be useful for our Javascript/Cordova client application.  The Cordova SDK has all of the features we will use, and we have a lot of JavaScript expertise to draw on.
+
+### iOS
+
+The iOS SDK has all of the features we need, and iOS has an Apollo GraphQL library as well.  Like Cordova and Android there is a lot of experience we can draw on for developing the app.
+
+
+## Data Model and API layer
+
+The following is a graphql schema for the proposed data model.  The GraphQL schema also includes the public APIs that will be implemented.
+
+### Sample Schema
+
+```graphql
+type Profile {
+  id: ID! @unique
+  email: String!
+  display_name: String!
+  biography: String!
+  pictureUrl: String!
+}
+
+type MemeList {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  memes: [Meme!]!
+}
+
+type Meme {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  photoUrl: String!
+  comments: [Comment!]!
+}
+
+type Comment {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  comment: String!
+}
+
+type Query {
+  profile(email: String!):Profile!
+  usersMemes(email: String!): [Meme!]!
+  feed(memeListId: ID!): [Meme!]!
+  meme(memeId: ID!):Meme
+}
+
+type Mutation {
+  createProfile(email: String!, display_name: String!, biography: String!, pictureUrl: String!):Profile!
+  updateProfile(id: ID!, email: String!, display_name: String!, biography: String!, pictureUrl: String!):Profile!
+  deleteProfile(id: ID!):Profile!
+
+  postMeme(picture: [Int!]!, topComment: String!,bottomComment: String!, memeListId: ID!): Meme!
+  deleteMeme(id: ID!): Meme!
+  likeMeme(id: ID!): Meme!
```

Comment:
Where we will store the Meme's likes?

Reply:
Hrm, should the like count be a value on the Meme or should we make it a "LIKE" object that has a id foreign key as well as a type so they can be for memes, comments, profiles, etc.

Reply:
I think like in the meme is not an option because how do will know if "I" already like it?